process/sync/baseSync.go

```diff
@@ -318,7 +303,7 @@ func (boot *baseBootstrap) ShouldSync() bool {
 }
 
 func (boot *baseBootstrap) removeHeaderFromPools(header data.HeaderHandler) []byte {
-	boot.headersNonces.Remove(header.GetNonce(), header.GetShardID())
+	boot.headers.RemoveHeaderByNonceAndShardId(header.GetNonce(), header.GetShardID())
```

Comment:
Please remove header by hash here in line 313 and remove this line. We now should remove headers targeted and not all the headers with the same nonce. We should analyzed when we want to remove all the headers. Here is not the case. We need to search after all calls of RemoveHeaderByNonceAndShardId method and do a carefully check.

Reply:
I did this search and is ok for now. It is called only here and in one method in blockProcessor where a cleanup behind final nonce is done, so is ok. This method remains the only place where we need to delete by hash.

Reply:
done.